controller/main_controller.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ *
+ * Collapsible Categories extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2015 phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\collapsiblecategories\controller;
+
+class main_controller implements main_interface
+{
+	/** @var \phpbb\collapsiblecategories\operator\operator */
+	protected $operator;
+
+	/** @var \phpbb\request\request */
+	protected $request;
+
+	/**
+	* Constructor
+	*
+	* @param \phpbb\collapsiblecategories\operator\operator $operator Operator object
+	* @param \phpbb\request\request                         $request  Request object
+	* @access public
+	*/
+	public function __construct(\phpbb\collapsiblecategories\operator\operator $operator, \phpbb\request\request $request)
+	{
+		$this->operator = $operator;
+		$this->request = $request;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function handle($forum_id)
+	{
+		// Throw an exception for non-AJAX requests or if the forum_id is missing
+		if (!$this->request->is_ajax() || $forum_id == 0)
```

Comment:
Can you update the docblock in the interface. The $forum_id param should now be "string" instead of "int".

Also `$forum_id == 0` is not working since it is a string. instead do `!$forum_id`